### PR TITLE
[MINOR] fix typo in documents

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

I use spell check tools checks typo in spark documents and fix them.
## How was this patch tested?

N/A

### CODE REVIEW

File: docs/streaming-programming-guide.md [hunk 1]
```diff
@@ -2037,7 +2037,7 @@ and configuring them to receive different partitions of the data stream from the
 For example, a single Kafka input DStream receiving two topics of data can be split into two
 Kafka input streams, each receiving only one topic. This would run two receivers,
 allowing data to be received in parallel, thus increasing overall throughput. These multiple
-DStreams can be unioned together to create a single DStream. Then the transformations that were
+DStreams can be united together to create a single DStream. Then the transformations that were
```
Comment (srowen) [inline docs/streaming-programming-guide.md]: This was not a typo. unioned is correct.

Comment (WeichenXu123) [inline docs/streaming-programming-guide.md]: @srowen recover 'unioned', thanks!

**Comment by srowen** [comment]
OK, did you survey all the docs? would be great to make a pass of everything, if you haven't already.

**Comment by WeichenXu123** [comment]
@srowen Yes, I check each md files, and I think it is done.

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**